src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -268,6 +268,8 @@ internal static unsafe bool IsPinnable(object? obj)
         /// Returns the HInstance for this module.  Returns -1 if the module doesn't have
         /// an HInstance.  In Memory (Dynamic) Modules won't have an HInstance.
         /// </summary>
+        [RequiresAssemblyFiles("Windows only assigns HINSTANCE to assemblies loaded from disk. " +
```

Comment:
This attribute needs to be duplicated in ref file and on other implementations of `GetHINSTANCE`.

Reply:
Right - I added it to the ref assembly, but I can't find any other implementation. Mono doesn't have this method at all (I guess since it's Windows only).